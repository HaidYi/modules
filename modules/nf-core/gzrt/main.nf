process GZRT {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/gzrt:0.9.1--h577a1d6_0':
        'biocontainers/gzrt:0.9.1--h577a1d6_0' }"

    input:
    tuple val(meta), path(fastqgz)

    output:
    tuple val(meta), path("${prefix}.fastq.gz"), emit: recovered
    path "versions.yml"                        , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    if (fastqgz.extension != "gz") {
        error "GZRT works with .gz files only."
    }

    prefix = task.ext.prefix ?: "${meta.id}_recovered"
    if ("${fastqgz}" == "${prefix}.fastq.gz") error "Input and output names are the same, set prefix in module configuration to disambiguate!"
    """
    gzrecover -p ${fastqgz} | gzip > ${prefix}.fastq.gz

    soft_line="${task.process}"
    ver_line="gzrt: \$(gzrecover -V |& sed '1!d ; s/gzrecover //')"
    cat <<-END_VERSIONS > versions.yml
    "\${soft_line}":
        \${ver_line}
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}_recovered"
    if ("${fastqgz}" == "${prefix}.fastq.gz") error "Input and output names are the same, set prefix in module configuration to disambiguate!"
    """
    echo "" | gzip > ${prefix}.fastq.gz

    soft_line="${task.process}"
    ver_line="gzrt: \$(gzrecover -V |& sed '1!d ; s/gzrecover //')"

    cat <<-END_VERSIONS > versions.yml
    "\${soft_line}":
        \${ver_line}
    END_VERSIONS
    """
}
