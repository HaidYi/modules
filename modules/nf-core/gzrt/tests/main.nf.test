nextflow_process {

    name "Test Process GZRT"
    script "../main.nf"
    process "GZRT"

    tag "modules"
    tag "modules_nfcore"
    tag "gzrt"

    test("Run gzrt on test2_1_corrupted - fastq.gz") {

        when {
            process {
                // file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test2_1_corrupted_10kb.fastq.gz', checkIfExists: true)
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test2_1.fastq.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("Run gzrt on test2_1_corrupted - fastq.gz - stub") {
        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test2_1.fastq.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}