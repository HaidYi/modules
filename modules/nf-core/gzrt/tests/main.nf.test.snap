{
    "Run gzrt on 30 paired-end reads - fastq.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_recovered.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b9467d4cb860eb2941a2078a3dd3cf41"
                ],
                "fastqrecovered": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_recovered.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b9467d4cb860eb2941a2078a3dd3cf41"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-27T19:02:56.484798"
    },
    "Run gzrt on 30 paired-end reads - fastq.gz - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_recovered.fastq.gz:md5,0f8202ea19c8ce3f69f4413bf139b29c"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b9467d4cb860eb2941a2078a3dd3cf41"
                ],
                "fastqrecovered": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_recovered.fastq.gz:md5,0f8202ea19c8ce3f69f4413bf139b29c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b9467d4cb860eb2941a2078a3dd3cf41"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-27T19:03:05.999888"
    }
}